### PR TITLE
Support for spymaster reset, fixes for spymaster logic, cleanup of unneeded event, tests

### PR DESCRIPTION
**Pull request by lazeratops**
This PR:

* Adds support for a new spymaster taking over if a spymaster leaves
  * https://github.com/daily-demos/modern-wordfare/issues/4
* Adds some other fixes for spymaster logic
  * Eg, unreliable spymaster button show due to typo
* Adds some more tests
  * For memstore, orchestrator, game classes
* Removes unneeded event
  * The leave game request the client was sending to the server is not necessary, since we already handle a socket disconnect as a game leave and leaving the game triggers that.
* Ensures consistent socket mapping storage
  * Socket mapping was not being stored when a player becomes a spymaster without having joined the team as a player before, meaning them leaving was not being properly registered at a later point.

### CODE REVIEW

File: src/client/game/board.ts [hunk 1]
```diff
@@ -165,13 +165,26 @@ export class Board {
...
 
     this.createTile(p, team, force);
     this.updateGameStatus();
     this.updateInteraction();
     toggleEndTurnButton(this.currentTurn, this.team);
   }
 
+  eject(playerID: string) {
+    removeTile(playerID);
+    if (this.spymasters.team1 === playerID) {
```
Comment (lazeratops) [inline src/client/game/board.ts:179]: Previously we were just calling `removeTile()` here. But what if a spymaster is leaving? We need to update the relevant join buttons to allow others to join the team as a spymaster if this is the case, hence this addition.

File: src/client/game/game.ts [hunk 2]
```diff
@@ -21,14 +21,13 @@ import {
...
   TurnResultData,
-  leaveGameEventName,
   restartGameEventName,
   RestartGameData,
   gameRestartedEventName,
   GameRestartedData,
   endTurnEventName,
   EndTurnData,
-  playerLeftgameEventName,
+  playerLeftGameEventName,
```
Comment (lazeratops) [inline src/client/game/game.ts:30]: Just fixing the casing on this event var name

File: src/client/game/nav.ts [hunk 3]
```diff
@@ -114,7 +114,7 @@ export function hideSpymasterBtn(team: Team) {
 }
 
 export function showSpymasterBtn(team: Team) {
-  if (team || team === Team.None) return;
+  if (!team || team === Team.None) return;
```
Comment (lazeratops) [inline src/client/game/nav.ts:117]: This was... pretty bad :/

**Review by jayne-mast — APPROVED** [review]
Great that you've added those extra tests!

**Pull request merged**